docs/mathematics/prime-number.mdx

```diff
@@ -0,0 +1,123 @@
+---
+title: Prime Number
+slug: /mathematics/prime-number
+---
+
+## Introduction
+
+Prime numbers are the positive integers having only two factors, 1 and the integer itself.
+
+## Explanation
+
+If p is a prime, then it’s only factors are necessarily 1 and p itself. Any number which does not follow this is termed as composite numbers, which means that they can be factored into other positive integers.
+#### Quick Example : 
+The first ten primes are 2, 3, 5, 7, 11, 13, 17, 19, 23, 29.
+
+Here 2 is prime as it's factors are 1 and 2 itself. Similarly 3 and 5 are prime as their factors are 1 and 3 and 1 and 5 respectively.
+**Note:** It should be noted that 1 is a non-prime number.
+
+## Algorithm
+
+ - Input the number `num`.
+ - Initialize `Flag` to 1.
+ - Check the number of factors of `num`. If factor is found then `Flag` is 0.
+ - Check the condition: if `Flag` is 1 then `num` is `prime` else `num` is `not prime`.
+
+## Code 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="c"
+  values={[
+    { label: 'C', value: 'c', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```c
+#include <stdio.h>
+#include <math.h>
+ // Function to check whether the Number is Prime Number or Not.
+int is_prime(int num) {
+  int Flag = 1;
+  for (int i = 2; i <= sqrt(num); i++) {
+    if (num % i == 0) {
+      Flag = 0;
+      break;
+    }
+  }
+  if (Flag == 1)
+    return 1;
+  else
+    return 0;
+}
+int main() {
+  printf("Enter the number: ");
+  int num;
+  scanf("%d", & num);
+  if (is_prime(num) == 1)
+    printf("%d is a Prime Number.", num);
+  else
+    printf("%d is not a Prime Number.", num);;
+  return 0;
+}
+```
+</TabItem>
+
+<TabItem value="java">
+
+```java
+import java.io.*;
+import java.util.*;
+public class Primenumber {
+    static boolean isPrime(int num) {
+        int Flag = 1;
+        for (int i = 2; i < Math.sqrt(num); i++) {
+            if (num % i == 0) {
+                Flag = 0;
+                break;
+            }
+        }
+        if (Flag == 1)
+            return true;
+        else
+            return false;
+    }
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.print("Enter the number: ");
+        int num = sc.nextInt();
+        if (isPrime(num))
+            System.out.println(num + " is an Prime Number.");
+        else
+            System.out.println(num + " is not an Prime Number.");
+    }
+}
+```
+</TabItem>
+</Tabs>
+
+## Sample Input and Output 
+#### Input :
+```
+Enter the number: 23
+```
+
+#### Output :
+```
+23 is a Prime Number.
+```
+#### Input :
+```
+Enter a number: 14
+```
+#### Output :
+```
+14 is not a Prime Number.
+```
+## Complexity Analysis
+- Time Complexity: `O(n)`
+- Space Complexity: `O(1)`
+
```

Comment:
Add credits for the code if you are picking from NeoAlgo

Reply:
No credits